src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private static final double DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final int MAX_GEOMETRIC_PROPERTY_SIZE = 10;
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String randomColor = colorSupplier.getRandomColor();
+        double circleRadius = getRandomValue();
+        double firstRightTriangleLeg = getRandomValue();
+        double secondRightTriangleLeg = getRandomValue();
+        double squareSide = getRandomValue();
+        double firstSideIsoscelesTrapezoid = getRandomValue();
+        double secondSideIsoscelesTrapezoid = getRandomValue();
+        double heightIsoscelesTrapezoid = getRandomValue();
+        double rectangleLength = getRandomValue();
+        double rectangleWidth = getRandomValue();
```

Comment:
generate right in figures constructors, where you need it. You end up not using most of what you've generated in advance
If case statements become too bulky - you can extract their logic to dedicated private methods